build.gradle.kts

```diff
@@ -1,226 +1,193 @@
+
+        // Plugin Dependencies. Uses `platformPlugins` property from the gradle.properties file for plugin from JetBrains Marketplace.
+        plugins(providers.gradleProperty("platformPlugins").map { it.split(',') })
 
-    create("integrationTest") {
-        java.srcDir("src/it/java")
-        resources.srcDir("src/it/resources")
-        compileClasspath += sourceSets.main.get().output
-        compileClasspath += configurations.testImplementation.get()
-        runtimeClasspath += compileClasspath + sourceSets.test.get().output
+        instrumentationTools()
+        pluginVerifier()
+        zipSigner()
+        testFramework(TestFrameworkType.Platform)
     }
-}
 
-// Dependencies are managed with Gradle version catalog - read more: https://docs.gradle.org/current/userguide/platforms.html#sub:version-catalog
-dependencies {
-    implementation("org.zeroturnaround:zt-zip:1.14")
-    implementation("org.jsoup:jsoup:1.17.1")
+    val lsp4jVersion = providers.gradleProperty("lsp4jVersion").get()
     implementation("org.eclipse.lsp4j:org.eclipse.lsp4j:$lsp4jVersion")
     implementation("org.eclipse.lsp4j:org.eclipse.lsp4j.debug:$lsp4jVersion")
-    // Required by lsp4j as the version from IJ is incompatible
-    implementation("com.google.code.gson:gson:2.10.1")
+
+    val flexmarkVersion = providers.gradleProperty("flexmarkVersion").get()
     implementation("com.vladsch.flexmark:flexmark:$flexmarkVersion")
     implementation("com.vladsch.flexmark:flexmark-ext-tables:$flexmarkVersion")
     implementation("com.vladsch.flexmark:flexmark-ext-autolink:$flexmarkVersion")
     implementation("com.vladsch.flexmark:flexmark-ext-gfm-strikethrough:$flexmarkVersion")
+
     testImplementation("com.redhat.devtools.intellij:intellij-common-ui-test-library:0.2.0")
     testImplementation("org.assertj:assertj-core:3.19.0")
-
 }
 
-// Set the JVM language level used to build the project. Use Java 11 for 2020.3+, and Java 17 for 2022.2+.
-kotlin {
-    jvmToolchain {
-        languageVersion = JavaLanguageVersion.of(17)
-        vendor = JvmVendorSpec.JETBRAINS
-    }
-}
+// Configure IntelliJ Platform Gradle Plugin - read more: https://plugins.jetbrains.com/docs/intellij/tools-intellij-platform-gradle-plugin-extension.html
+intellijPlatform {
+    pluginConfiguration {
+        version = providers.gradleProperty("pluginVersion")
 
-// Configure Gradle IntelliJ Plugin - read more: https://plugins.jetbrains.com/docs/intellij/tools-gradle-intellij-plugin.html
-intellij {
-    pluginName = properties("pluginName")
-    version = properties("platformVersion")
-    type = properties("platformType")
-    updateSinceUntilBuild = false
-    // Plugin Dependencies. Uses `platformPlugins` property from the gradle.properties file.
-    plugins = properties("platformPlugins").map { it.split(',').map(String::trim).filter(String::isNotEmpty) }
-}
+        // Extract the <!-- Plugin description --> section from README.md and provide for the plugin's manifest
```

Comment:
@angelozerr @fbricon below is an addition to use README and CHANGELOG files contents to populate plugin description, made manually for now. is it something we want to do ?